buildSrc/src/main/groovy/airbyte-docker.gradle

```diff
@@ -125,6 +129,39 @@ class AirbyteDockerTaskFactory {
         return map
     }
 
+    // Query all docker images at most once for all tasks, at task creation time.
+    static def lazyImageHashesAtTaskCreationTime = new LazyImageHashesCache()
+
+    static class LazyImageHashesCache {
+        private Map<String, String> lazyValue
+
+        synchronized Map<String, String> get(Project project) {
+            if (lazyValue == null) {
+                lazyValue = collectKnownImageHashes(project)
+            }
+            return lazyValue
+        }
+    }
+
+    // Global mapping of tagged image name to gradle project.
+    // This is populated at configuration time and accessed at task creation time.
+    // All keys verify isTaggedImageOwnedByThisRepo.
+    static def taggedImageToProject = new ConcurrentHashMap<String,Project>()
+
+    static boolean isTaggedImageOwnedByThisRepo(String taggedImage) {
+        if (!taggedImage.startsWith("airbyte/")) {
+            // Airbyte's docker images are all prefixed like this.
+            // Anything not with this prefix is therefore not owned by this repo.
+            return false
+        }
+        if (taggedImage.startsWith("airbyte/base-airbyte-protocol-python:")) {
```

Comment:
For my information: do we know where this is built?

Reply:
If you did not find any reference to this image its probably a legacy image that's not in use anymore.

Reply:
I did as a base image in `base-normalization`. That doesn't mean it's actually in use, IDK.

Reply:
If  `taggedImage` refers to a local image in this context we're safe to remove this statement.

Reply:
As it turns out, `airbyte/base-airbyte-protocol-python` is used for `airbyte/normalization-tidb` which in turn is used to build `airbyte/destination-tidb`. This might be a problem.